app/Http/Controllers/SupplyController.php

```diff
@@ -220,31 +219,32 @@ public function logNetworkView(Request $request, Adselect $adselect, $id)
 
         $logIp = bin2hex(inet_pton($request->getClientIp()));
 
-        $cid = Utils::getRawTrackingId($request->query->get('cid'));
-        $tid = Utils::getRawTrackingId($request->cookies->get('tid')) ?: $logIp;
+        $impressionId = $request->query->get('iid');
+        $eventId = Utils::getRawTrackingId(Utils::createTrackingId(config('app.adserver_secret'), $impressionId));
+        $trackingId = Utils::getRawTrackingId($request->cookies->get('tid')) ?: $logIp;
         $payFrom = $request->query->get('pfr');
+        $payTo = $request->query->get('pto');
```

Comment:
Payment_to (pto) should be taken from config instead of query.

Reply:
done